AnkiDroid/src/test/java/com/ichi2/anki/dialogs/CreateDeckDialogTest.java

```diff
@@ -0,0 +1,119 @@
+
+@RunWith(RobolectricTestRunner.class)
+public class CreateDeckDialogTest extends RobolectricTest {
+
+    private ActivityScenario<DeckPicker> mActivityScenario;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        mActivityScenario = ActivityScenario.launch(DeckPicker.class);
+        mActivityScenario.moveToState(Lifecycle.State.STARTED);
+    }
+
+    @Test
+    public void testCreateFilteredDeckFunction() {
+        ensureCollectionLoadIsSynchronous();
+
+        mActivityScenario.onActivity(activity -> {
+            AtomicReference<Boolean> isCreated = new AtomicReference<>(false);
+            String deckName = "filteredDeck";
+            shadowOf(getMainLooper()).idle();
+
+            activity.mCreateDeckDialog.setOnNewDeckCreated((id) -> {
+                // a deck was created
+                try {
+                    isCreated.set(true);
+                    assertThat(id, is(activity.getCol().getDecks().id(deckName)));
+                } catch (FilteredAncestor filteredAncestor) {
+                    filteredAncestor.printStackTrace();
+                }
+            });
+
+            activity.mCreateDeckDialog.createFilteredDeck(deckName);
+            assertThat(isCreated.get(), is(true));
+        });
+    }
+
+    @Test
+    public void testCreateSubDeckFunction() throws FilteredAncestor{
+        ensureCollectionLoadIsSynchronous();
+        Long deckParentId = new AnkiActivity().getCol().getDecks().id("Deck Name");
+
+        mActivityScenario.onActivity(activity -> {
+            AtomicReference<Boolean> isCreated = new AtomicReference<>(false);
+            String deckName = "filteredDeck";
+            shadowOf(getMainLooper()).idle();
+
+            activity.mCreateDeckDialog.setOnNewDeckCreated((id) -> {
+                try {
+                    isCreated.set(true);
+                    String deckNameWithParentName = activity.getCol().getDecks().getSubdeckName(deckParentId, deckName);
+                    assertThat(id, is(activity.getCol().getDecks().id(deckNameWithParentName)));
+                } catch (FilteredAncestor filteredAncestor) {
+                    filteredAncestor.printStackTrace();
+                }
+            });
+
+            activity.mCreateDeckDialog.createSubDeck(deckParentId, deckName);
+            assertThat(isCreated.get(), is(true));
+        });
+    }
+
+    @Test
+    public void testCreateDeckFunction() throws FilteredAncestor{
+        ensureCollectionLoadIsSynchronous();
+
+        mActivityScenario.onActivity(activity -> {
+            AtomicReference<Boolean> isCreated = new AtomicReference<>(false);
+            String deckName = "Deck Name";
+            shadowOf(getMainLooper()).idle();
+
+            activity.mCreateDeckDialog.setOnNewDeckCreated((id) -> {
+                // a deck was created
+                try {
+                    isCreated.set(true);
+                    assertThat(id, is(activity.getCol().getDecks().id(deckName)));
```

Comment:
Also for the other call

For legacy reasons, `id` will create the deck, but when passing in a string, it'll return the `default deck` if it's not found. If for some reason `id` is the default deck, then this would pass

```suggestion
                    assertThat(id, is(activity.getCol().getDecks().byName(deckName).getLong("id")));
```